utils/dial/dial.go

```diff
@@ -0,0 +1,88 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+// Package dial provides gRPC connection setup for Percona Platform.
+package dial
+
+import (
+	"context"
+	"net"
+	"time"
+
+	"github.com/percona/pmm/utils/tlsconfig"
+	"github.com/percona/pmm/version"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+const (
+	platformAuthType = "PP-1"
+	dialTimeout      = 10 * time.Second
+)
+
+type platformAuth struct {
```

Comment:
The whole type could probably be in a separate small file

Reply:
Done